### PR TITLE
Adds radial support + balloon alerts to stack crafting, cult metal now uses a radial when making buildings

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Allows stack items to use a radial when crafting.
- Changes runed metal / cult buildings to use a radial.
  -~~As a consequence, this disallows people from making runed toilets and the other useless stuff, though it can probably be put on the second side wheel - still thinking on that, not much is lost imo~~

image

*radius subject to change*

## Why It's Good For The Game

Radial menus are a bit more pleasing to use than a long list. 
Additionally, this allows for the description / guides for the cult buildings to be put in the info button (`(?)`).

## Changelog

:cl: Melbert
code: Adds radial support to stack crafting.
qol: Cult buildings are now made via radial, instead of stack window.
qol: Stack craftingnow uses balloon alerts.
/:cl:

### CODE REVIEW

**Comment by Colovorat** [comment]
noooo my runic toilet 😭

**Comment by MrMelbert** [comment]
image

nevermind, i found a decent way to keep the toilet in

**Comment by Fikou** [comment]
not a fan of it being inconsistent whether its a tgui list or radials

**Comment by MrMelbert** [comment]
> not a fan of it being inconsistent whether its a tgui list or radials

Radial support can be added to more and more stack types which have more notable recipes, that's why I added it as a base feature of stacks

File: code/game/objects/items/stacks/sheets/runed_metal.dm [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+/obj/item/stack/sheet/runed_metal/interact(mob/user)
+	if(!IS_CULTIST(user))
+		to_chat(user, span_warning("Only one with forbidden knowledge could hope to work this metal..."))
+		return FALSE
+
+	var/turf/user_turf = get_turf(user)
+	var/area/user_area = get_area(user)
+
+	var/is_valid_turf = user_turf && (is_station_level(user_turf.z) || is_mining_level(user_turf.z))
+	var/is_valid_area = user_area && (user_area.area_flags & CULT_PERMITTED)
```
Comment (JohnFulpWillard) [inline code/game/objects/items/stacks/sheets/runed_metal.dm:92]: is the first check even necessary? area_flags includes both levels

File: code/game/objects/items/stacks/stack.dm [hunk 2]
```diff
@@ -250,64 +282,149 @@
...
+	if(QDELETED(created))
+		return TRUE
+
+	// Add fingerprints first, otherwise created might already be deleted because of stack merging
+	created.add_fingerprint(builder)
+	if(isitem(created))
+		builder.put_in_hands(created)
+
+	//BubbleWrap - so newly formed boxes are empty
+	if(istype(created, /obj/item/storage))
```
Comment (JohnFulpWillard) [inline code/game/objects/items/stacks/stack.dm:422]: not related to your PR but this is a very weird way of handling this.

File: code/game/objects/items/stacks/stack.dm [hunk 3]
```diff
@@ -359,21 +474,23 @@
 				if(!window_structure.fulltile)
 					continue
 			if(object.density || NO_BUILD & object.obj_flags)
-				to_chat(usr, span_warning("There is \a [object.name] here. You can\'t make \a [recipe.title] here!"))
+				builder.balloon_alert(builder, "there's something in the way!")
```
Comment (JohnFulpWillard) [inline code/game/objects/items/stacks/stack.dm:477]:
```suggestion
				builder.balloon_alert(builder, "something is in the way!")
```

File: code/game/objects/items/stacks/stack.dm (hunk 2 above)
Comment (MrMelbert) [reply code/game/objects/items/stacks/stack.dm:422]: Yeah I saw this code block when I moved it over and thought that was jank as hell. 
Future refactor though.

File: code/game/objects/items/stacks/sheets/runed_metal.dm (hunk 1 above)
Comment (MrMelbert) [reply code/game/objects/items/stacks/sheets/runed_metal.dm:92]: In certain events where a cult permitted area ends up in deep space, such as survival pods and shuttles

Comment (JohnFulpWillard) [reply code/game/objects/items/stacks/sheets/runed_metal.dm:92]: i didnt even know survival pods were allowed, i wonder what the reason behind it is. not for this pr though

**Pull request merged**